src/main.c

```diff
@@ -3742,6 +3785,46 @@ static void create_per_thread_listeners(void)
     }
 }
 
+H2O_NORETURN static void *h2olog_thread(void *_ctx)
+{
+    while (1) {
+        int fd = accept(conf.h2olog.listening_fd, NULL, 0);
+        if (fd == -1) {
+            continue;
+        }
+        if (fcntl(fd, F_SETFD, FD_CLOEXEC) != 0) {
```

Comment:
I'm afraid we cannot accept a socket then set the CLOEXEC flag. The problem is than when fork is called by a different thread while we are between `accept` and `fcntl`, the socket is leaked to the child process being forked, and probably never be closed.

The correct approach would be to set the listening socket to non-blocking mode, call `select`, and if the listening socket is read ready, call `cloexec_accept`.